src/containers/WrongChainRecovery/WrongChainRecovery.tsx

```diff
@@ -1,4 +1,151 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { CoinsSelectAutocomplete } from '~/components/CoinsSelectAutocomplete';
+import { useAlertBanner } from '~/contexts';
+import { mapSdkErrorToAlert, safeEnv } from '~/helpers';
+import {
+  allCoinMetas,
+  CoinMetadata,
+  wrongChainRecoveryCoins,
+} from '~/helpers/config';
+import { WrongChainRecoveryForm } from './WrongChainRecoveryForm';
+
 export function WrongChainRecovery() {
-  // TODO(BG-67209)
-  return <>{'Coming Soon'}</>;
+  const { env } = useParams<'env'>();
+  const environment = safeEnv(env);
+  const [, setAlert] = useAlertBanner();
+
+  const sourceCoins: readonly CoinMetadata[] = Object.keys(
+    wrongChainRecoveryCoins[environment]
+  ).map(key => {
+    return allCoinMetas[key as keyof typeof allCoinMetas];
+  });
+  const [sourceCoin, setSourceCoin] = useState<string>(sourceCoins[0].value);
+
+  const destinationCoins = useMemo(() => {
+    return wrongChainRecoveryCoins[environment][sourceCoin];
+  }, [environment, sourceCoin]);
+  const [destinationCoin, setDestinationCoin] = useState<string>(
+    destinationCoins[0].value
+  );
+
+  useEffect(() => {
+    const newDestinationCoins =
+      wrongChainRecoveryCoins[environment][sourceCoin];
+    if (
+      newDestinationCoins.length &&
+      !newDestinationCoins.find(item => item.value === destinationCoin)
+    ) {
+      setDestinationCoin(newDestinationCoins[0].value);
+    }
```

Comment:
why do you need to use `newDestinationCoins` , wouldn't `destinationCoins` be refreshed if `environment` or `sourceCoin` change anyways?

Reply:
Good catch, this was left-over before I did the useMemo for destinationCoins.

Reply:
Fixed